src/app/shared/auth-nav-menu/user-menu/user-menu.component.ts

```diff
@@ -108,6 +122,21 @@ export class UserMenuComponent implements OnInit {
     // set user
     this.user$ = this.authService.getAuthenticatedUserFromStore();
 
+    // Retrieve potential profile id using user id.
+    this.user$.pipe(
+      filter(isNotEmpty),
+      switchMap(
+        user => this.researcherProfileService.findById(user.id, false, true, followLink('item')).pipe(
+          getAllCompletedRemoteData(),
+          getRemoteDataPayload(),
+          switchMap((researcherProfile) => this.researcherProfileService.findRelatedItemId(researcherProfile))
+        )
+      ),
+    ).subscribe((profileId: String) => {
+      if (isNotEmpty(profileId)) {
```

Comment:
Can you move this condition statement as the last element of the the pipe section (as L127 but just after L133)